extension/src/target.ts

```diff
@@ -48,6 +55,116 @@ function generateRoleSelector(
   return selector
 }
 
+function generateAltTextSelector(element: Element): string | undefined {
+  if (
+    element instanceof HTMLImageElement === false &&
+    element instanceof HTMLAreaElement === false &&
+    element instanceof HTMLInputElement === false
+  ) {
+    return undefined
+  }
+
+  if (element instanceof HTMLInputElement && element.type !== 'image') {
+    return undefined
+  }
+
+  const alt = element.alt.trim()
+
+  if (alt === '') {
+    return undefined
+  }
+
+  const matches = queryAllByAltText(document.body, alt, { exact: true })
+
+  if (matches.length !== 1) {
+    return undefined
+  }
+
+  return alt
+}
+
+function generateLabelSelector(
+  element: Element,
+  { labels }: AriaDetails
+): string | undefined {
+  if (element instanceof HTMLElement === false) {
+    return undefined
+  }
+
+  for (const label of labels) {
+    const trimmed = label.trim()
+
+    const matches = queryAllByLabelText(document.body, trimmed, {
+      exact: true,
+    })
+
+    if (!matches.includes(element)) {
+      continue
+    }
+
+    if (matches.length > 1) {
+      continue
+    }
+
+    return trimmed
+  }
+
+  return undefined
+}
+
+function generatePlaceholderSelector(element: Element): string | undefined {
+  if (
+    element instanceof HTMLInputElement === false &&
+    element instanceof HTMLTextAreaElement === false
+  ) {
+    return undefined
+  }
+
+  const placeholder = element.placeholder.trim()
+
+  if (placeholder === '') {
+    return undefined
+  }
+
+  const matches = queryAllByPlaceholderText(document.body, placeholder, {
+    exact: true,
+  })
```

Comment:
I'm a bit on the fence about using exact matching. It's the most reliable for generating selectors, but it's not necessarily the most reliable when authoring tests. I think in general you want to be pretty loose so that you don't have to take e.g. alt texts in the accessible name into account.

One way is to first do a loose match and if that doesn't produce a unique result do an exact match. Anyway, just some food for thought.